actionview/lib/action_view/helpers/translation_helper.rb

```diff
@@ -172,8 +140,47 @@ def scope_key_by_partial(key)
           end
         end
 
+        def html_escape_translation_options(options)
+          return options if options.empty?
+          html_safe_options = options.dup
+
+          options.each do |name, value|
+            unless TranslationHelper.i18n_option?(name) || (name == :count && value.is_a?(Numeric))
+              html_safe_options[name] = ERB::Util.html_escape(value.to_s)
+            end
+          end
+
+          html_safe_options
+        end
+
         def html_safe_translation_key?(key)
-          /(?:_|\b)html\z/.match?(key.to_s)
+          /(?:_|\b)html\z/.match?(key)
+        end
+
+        def html_safe_translation(translation)
+          if translation.respond_to?(:map)
+            translation.map { |element| element.respond_to?(:html_safe) ? element.html_safe : element }
+          else
+            translation.respond_to?(:html_safe) ? translation.html_safe : translation
+          end
+        end
+
+        def missing_translation(key, options)
+          keys = I18n.normalize_keys(options[:locale] || I18n.locale, key, options[:scope])
+
+          title = +"translation missing: #{keys.join(".")}"
+
+          options.each do |name, value|
+            unless name == :default || name == :scope
```

Comment:
If you go for https://github.com/rails/rails/pull/39989#discussion_r487587243:

```suggestion
            unless name == :scope
```